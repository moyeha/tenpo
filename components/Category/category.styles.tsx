import { Image, Text, View } from "react-native";
import styled from "styled-components";
import theme from "../../theme";

const MainContainer = styled(View)({
  // flex: 1,
  justifyContent: "center",
  alignItems: "center",
});

const TitleContainer = styled(View)({
  position: "absolute",
  justifyContent: "center",
  alignItems: "center",
});

const Title = styled(Text)({
  color: theme.colors.white,
  fontFamily: "Gotham-Black",
  fontSize: "16px",
  textAlign: "center",
  textTransform: "uppercase",
});

const StyledImage = styled(Image)({
  height: "80px",
  width: "160px",
  resizeMode: "contain",
  marginBottom: "8px",
});

export { TitleContainer, MainContainer, StyledImage, Title };
