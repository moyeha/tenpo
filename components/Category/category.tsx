import React from "react";
import { Category as CategoryType } from "../../store/categories/slice";
import {
  MainContainer,
  StyledImage,
  Title,
  TitleContainer,
} from "./category.styles";

type RestaurantProps = CategoryType;

const Category = (props: RestaurantProps) => {
  return (
    <MainContainer>
      {!!props?.imagePath && <StyledImage source={props?.imagePath} />}
      <TitleContainer>
        <Title>{props?.nombre}</Title>
      </TitleContainer>
    </MainContainer>
  );
};

export default Category;
