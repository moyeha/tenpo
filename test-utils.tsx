import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react-native";
import type { RenderOptions } from "@testing-library/react-native";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "./store";
import type { AppStore, RootState } from "./store";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { ThemeProvider } from "styled-components";
import theme from "./theme";

const initialMetrics = {
  frame: { height: 0, width: 0, x: 0, y: 0 },
  insets: { bottom: 0, left: 0, right: 0, top: 0 },
};

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  Element: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  const Wrapper = ({ children }: PropsWithChildren<{}>) => {
    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <SafeAreaProvider initialMetrics={initialMetrics}>
            {children}
          </SafeAreaProvider>
        </ThemeProvider>
      </Provider>
    );
  };
  return { store, ...render(Element, { wrapper: Wrapper, ...renderOptions }) };
}
