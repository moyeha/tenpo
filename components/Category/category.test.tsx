import { screen } from "@testing-library/react-native";
import React from "react";
import Category from "./category";
import { renderWithProviders } from "../../test-utils";

describe("Category", () => {
  test("Deberia estar el titulo Categorias", async () => {
    renderWithProviders(<Category id="1" nombre="nombre-categoria" imagePath={null} />);
    expect(screen.queryByText(/Categorias/i)).toMatchSnapshot();
  });
});
