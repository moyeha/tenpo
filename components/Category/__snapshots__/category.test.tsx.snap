// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Deberia estar el titulo Categorias 1`] = `null`;
